src/libos/src/config.rs

```diff
@@ -128,15 +126,23 @@ pub struct ConfigUntrustedUnixSock {
     pub libos: PathBuf,
 }
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub struct ConfigMount {
     pub type_: ConfigMountFsType,
     pub target: PathBuf,
     pub source: Option<PathBuf>,
     pub options: ConfigMountOptions,
 }
 
-#[derive(Debug, PartialEq)]
+#[derive(Clone, Debug)]
+pub struct ConfigApp {
+    pub entry_points: Vec<PathBuf>,
+    pub stage: String,
+    pub mount: Vec<ConfigMount>,
+    pub encrypted: bool,
```

Comment:
I can not see the benefit of this newly added field `encrypted`. 

The `init` program should pass `key`  to `mount_rootfs` syscall for the encrypted image, if the value is not passed or the key is wrong, it will fail.

So this filed in Occlum.json seems to be useless

Reply:
> encrypted

You are right about the "useless" . The reason I passed "encrypted" from build process to libos was trying to give user a way (checking the generared json file) to tell why the image could be boot successfully. But since no key, no successful boot for encrypted image, then perhaps it is useless.

Reply:
If we keep this filed, then maybe Occlum can check this value with key parameter and early return error before mounting FS.

Reply:
I will update the commit with key&type check in the entry of do_mount_rootfs for general boot case.